opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -389,31 +390,24 @@ private String getFirstRevision(String fullpath) throws IOException {
                 ensureCommand(CMD_PROPERTY_KEY, CMD_FALLBACK),
                 "rev-list",
                 "--reverse",
-                "--max-count=1",
                 "HEAD",
                 "--",
                 fullpath
         };
 
         Executor executor = new Executor(Arrays.asList(argv), new File(getDirectoryName()),
                 RuntimeEnvironment.getInstance().getInteractiveCommandTimeout());
-        int status = executor.exec();
-
-        try (BufferedReader in = new BufferedReader(
-                new InputStreamReader(executor.getOutputStream()))) {
-            String line;
+        HeadHandler headHandler = new HeadHandler(1);
+        int status = executor.exec(false, headHandler);
 
-            if ((line = in.readLine()) != null) {
-                return line.trim();
-            }
+        String line;
+        if (headHandler.size() > 0 && (line = headHandler.get(0)) != null) {
```

Comment:
Nit: I think your implementation does ensure non-null for a valid index. Maybe that could be stated firmly in the Javadoc?